Language/Generics/Superbounded_types/class_A03_t01.dart

```diff
@@ -7,9 +7,8 @@
 ///   ...
 /// - `T` is an immediate subterm of a redirecting factory constructor signature
 ///
-/// @description Checks that compile error is thrown when a parameterized
-/// super-bounded type `T` is used in a redirecting factory constructor
-/// signature
+/// @description Checks that it is a compile-time error if a super-bounded type
+/// is an immediate subterm of a redirecting factory constructor signature
```

Comment:
We have two errors here: (1) The redirection uses a super-bounded type (this is the one that we're trying to test), and (2) the redirection corresponds to a type which isn't assignable to the return type of the constructor, `A<X>`.

So we'd need to use `class A {...}` rather than `class A<X> {...}` in order to make the redirection type correct, such that the _only_ error is the fact that the redirection type is super-bounded.

Something like this should work:

```dart
class A {
  ...
}

class C<X extends C<X>> extends A {}
```

This implies that `C<S>` is a subtype of `A` for all `S`.

Reply:
Thank you. Fixed

Reply:
Thanks!